library/core/src/option.rs

```diff
@@ -90,6 +90,205 @@
+//! * [`as_pin_ref`] converts from [`&Pin`]`<Option<T>>` to `Option<`[`Pin`]`<&T>>`
+//! * [`as_pin_mut`] converts from [`&mut Pin`]`<Option<T>>` to `Option<`[`Pin`]`<&mut T>>`
+//!
+//! [`&mut Pin`]: crate::pin::Pin
+//! [`&Pin`]: crate::pin::Pin
+//! [`as_deref`]: Option::as_deref
+//! [`as_deref_mut`]: Option::as_deref_mut
+//! [`as_mut`]: Option::as_mut
+//! [`as_pin_ref`]: Option::as_pin_ref
+//! [`as_pin_mut`]: Option::as_pin_mut
+//! [`as_ref`]: Option::as_ref
+//! [`Pin`]: crate::pin::Pin
+//!
+//! ## Extracting the contained value
+//!
+//! These methods extract the contained value in an [`Option`] when it is
+//! the [`Some`] variant. If the [`Option`] is [`None`]:
+//!
+//! * [`expect`] panics with a provided custom message
+//! * [`unwrap`] panics with a generic message
+//! * [`unwrap_or`] returns the provided default value
+//! * [`unwrap_or_default`] returns the default value of the type `T`
+//!   (which must implement the [`Default`] trait)
+//! * [`unwrap_or_else`] evaluates a provided function
+//!
+//! [`Default`]: crate::default::Default
+//! [`expect`]: Option::expect
+//! [`unwrap`]: Option::unwrap
+//! [`unwrap_or`]: Option::unwrap_or
+//! [`unwrap_or_default`]: Option::unwrap_or_default
+//! [`unwrap_or_else`]: Option::unwrap_or_else
+//!
+//! ## Transforming contained values
+//!
+//! * [`map`] transforms [`Some<T>`] to [`Some<U>`] using the provided
+//!   function
+//! * [`map_or`] transforms [`Some<T>`] to a value of `U` using the
+//!   provided function, or transforms [`None`] to a provided default value
+//!   of `U`
+//! * [`map_or_else`] transforms [`Some<T>`] to [`Some<U>`] using the
+//!   provided function, or transforms [`None`] to a value of `U` using
+//!   another provided function
+//! * [`ok_or`] transforms [`Some(v)`] to [`Ok(v)`], and [`None`] to
+//!   [`Err(err)`] using the provided default `err` value
+//! * [`ok_or_else`] transforms [`Some(v)`] to [`Ok(v)`], and [`None`] to
+//!   a value of [`Err<E>`] using the provided function
+//!
+//! [`Err(err)`]: Err
+//! [`map`]: Option::map
+//! [`map_or`]: Option::map_or
+//! [`map_or_else`]: Option::map_or_else
+//! [`Ok(v)`]: Ok
+//! [`ok_or`]: Option::ok_or
+//! [`ok_or_else`]: Option::ok_or_else
+//! [`Some(v)`]: Some
+//!
+//! ## Boolean operators
+//!
+//! These methods treat the [`Option`] as a boolean value, where [`Some`]
+//! acts like [`true`] and [`None`] acts like [`false`]. There are two
+//! categories of these methods: ones that take an [`Option`] as input, and
+//! ones that take a function as input (to be lazily evaluated).
+//!
+//! The [`and`], [`or`], and [`xor`] methods take another [`Option`] as
+//! input, and produce an [`Option`] as output. Only the [`and`] method can
+//! produce an [`Option<U>`] value having a different inner type `U` than
+//! [`Option<T>`].
+//!
+//! | method  | self      | input     | output    |
+//! |---------|-----------|-----------|-----------|
+//! | [`and`] | N/A       | `None`    | `None`    |
```

Comment:
`N/A` is a weird placeholder for "don't care". In [IEEE1164 terms](https://en.wikipedia.org/wiki/IEEE_1164) `-` is used to indicate a don't care value. I would also suggest an asterisk `*`

Reply:
Also, the same is true the other way around, e.g. self is `None` and `input` is don't care

Reply:
Thanks! I might have accidentally deleted a line there.

I'm more used to seeing `X` as "don't care" in datasheets, but that might be easy to confuse with a variable `x`. I worry that a "-" might disappear too easily in a default browser font. Making it a monospaced `-` might help, but might also be confusing. Maybe I should center those columns, so a hyphen-dash won't disappear as easily? Using an em-dash seems possibly excessive, but I might consider it.

Reply:
I think I'm going to replace "N/A" with "(ignored)" or "(not provided)" as appropriate, especially because we're not trying to keep this a single character width like some datasheet truth tables are.